### PR TITLE
ldc: update to 1.38.0

### PR DESCRIPTION
**Pull request by jiegec**
Topic Description
-----------------

- ldc: fix build path
- ldc: update to 1.38.0
- ldc: mark loongson3,mips64r6el,ppc64el as FAIL_ARCH
- ldc: remove custom build script and use cmakeninja template
- ldc: switch back to mainline LLVM
- ldc: import and refresh patches
- ldc: (ppc64el) mark as FAIL_ARCH
    FIXME: LDC does not support IEEE Long Double on POWER.
- ldc: update to 1.37.0
    - Build against LLVM 17 (llvm-17).
    - Lint scripts in accordance with the Styling Manual.
    - Rebase patches against 1.37.0.
    - Track patches at AOSC-Tracking/ldc @ aosc/v1.37.0.

Package(s) Affected
-------------------

- ldc: 1.38.0
- liblphobos: 1.38.0

Security Update?
----------------

No

Build Order
-----------

```
#buildit ldc:+stage2 ldc
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by jiegec** [comment]
Can compile and run example code from dlang.org.

**Comment by jiegec** [comment]
Tested on amd64 & loongarch64

File: lang-dlang/ldc/01-liblphobos/prepare.stage2 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+GDMD_WRAPPER="https://cdn.jsdelivr.net/gh/D-Programming-GDC/gdmd@ff2c97a47408fb71c18a2d453294d18808a97cc5/dmd-script"
+abinfo "Downloading gdmd wrapper ..."
```
Comment (Cyanoxygen) [inline lang-dlang/ldc/01-liblphobos/prepare.stage2]: Please let ACBS handle this download.

**Review by Cyanoxygen — DISMISSED** [review]

File: lang-dlang/ldc/01-liblphobos/prepare.stage2 (hunk 1 above)
Comment (jiegec) [reply lang-dlang/ldc/01-liblphobos/prepare.stage2]: Since #6071 is merged, I will use that instead.

Comment (jiegec) [reply lang-dlang/ldc/01-liblphobos/prepare.stage2]: Done.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**